# Personal Project: Hospital Provider Cost Reports

## Exploratory Data Analysis on Hospital Provider Cost Reports

### Introduction
This project analyzes **Hospital Provider Cost Reports** published between 2011 and 2022. The dataset includes annual hospital cost reporting files, which were combined into a single consolidated dataset for analysis.

Our primary objective is to perform a comprehensive analysis of hospital provider cost report data to identify patterns, trends, and potential areas for improvement. This work will serve as a learning exercise to strengthen my data analysis skills and will result in a polished portfolio piece showcasing my ability to transform raw data into actionable insights.

📂 **Data Source**: [CMS Hospital Provider Cost Report](https://data.cms.gov/provider-compliance/cost-report/hospital-provider-cost-report)

### Research Questions
The following questions guide the analysis of hospital cost report data, with emphasis on financial performance, capacity, and care delivery trends across hospitals and regions.  

Financial Performance
- How has hospital net income changed over time, both nationally and by state?
- What is the average cost-to-charge ratio across hospitals, and which hospitals are outliers?
- How has the cost-to-charge ratio evolved over the years?
- How do urban vs. rural hospitals differ in financial performance trends?
- What are the long-term trends in assets, liabilities, and fund balances?

Capacity & Operations
- Which states or counties have the highest hospital capacity in terms of beds and patient days?
- How has staffing (FTE employees) changed relative to patient volume?

Care Delivery & Equity
- What are the trends in charity care and uncompensated care over time?
- How much uncompensated care is provided by region or hospital type?
- How do inpatient vs. outpatient revenues vary by hospital size and type?

### Who is the intended audience?
The insights will be tailored to meet the needs of several stakeholder groups:
- Hospital administrators and executives – to inform operational and financial decision-making
- Researchers & analysts – to serve as a reference point for further studies
- General public & patient advocates – to promote understanding of healthcare costs and trends
- Policy makers and government agencies – to guide regulation, funding allocation, and healthcare reform initiatives

### Type of Analyses
Descriptive analysis:
- Visualize cost distribution by department and facility characteristics
- Analyze Medicare settlement trends and utilization data
- Explore patterns that might indicate cost drivers or inefficiencies

Financial insights:
- Highlight key KPIs like cost-to-reimbursement ratios
- Detect and flag unusual cost spikes or trends in expenses
- Provide easy-to-understand summaries for quick decision-making

Marketing/strategy angle:
- Benchmark facilities against each other on cost efficiency
- Show utilization strengths in high-demand services
- Present geographic and service-line heatmaps for competitive insights

### Preparing the Environment

We will import the required libraries and read in the data set.
- Pandas - Data Manipulation
- Matplotlib - Data Visualization
- Seaborn - Data Visualization
- Warnings - Utility usage libraries

In [1]:
# Import libraries and alias for easy reading
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings

In [6]:
# Suppress all warning messages to keep the output clean
warnings.filterwarnings('ignore')

# Configure plot styles: use matplotlib's default style
plt.style.use("default")

# Apply seaborn's theme for better-looking plots
sns.set_theme()

# Display all columns when printing DataFrames
pd.set_option("display.max_columns", None)

# Set the number of decimal places to display in DataFrames
pd.set_option("display.precision", 2)

In [15]:
# File path for the CSV data (replace with your own path if needed)
file_path = "C:/Users/Jason/Documents/Documents/Projects/Hospital_Provider_Cost_Report/Data/interim/hospital_provider_cost_output.csv"

# Read/load data in CSV format into a pandas DataFrame for analysis
hospital_provider_cost_data = pd.read_csv(file_path)

### Data Source & Preparation

The dataset comes from annual Hospital Provider Cost Reports, published between 2011 and 2022. Each year was originally provided as a separate CSV file (such as `CostReport_2011_Final.csv`, `CostReport_2012_Final.csv`, `CostReport_2022_Final.csv`).  

To simplify the analysis, these files were combined into a **single consolidated dataset**.  
A preprocessing script (`merge_hospital_provider_cost_report_data.py`) was used to:
- Load all CSV files from 2011–2022
- Extract the year from the filename
- Add a new column `Cost Report Year`
- Merge them into one unified dataset
- Save the result as `hospital_provider_cost_output.csv`

This consolidated file is what we use for all further analysis.

### Data Exploration
Let's have a look at the data using df.head() and df.tail() function.

In [19]:
# Display the first 5 rows of the dataset
hospital_provider_cost_data.head()

rpt_rec_num  Provider CCN                        Hospital Name  \
0          285         10005      MARSHALL MEDICAL CENTER - SOUTH   
1         1022        271326            BEARTOOTH BILLINGS CLINIC   
2         1496         10052       LAKE MARTIN COMMUNITY HOSPITAL   
3         1501         13025       HEALTHSOUTH LAKESHORE HOSPITAL   
4         1504        103037  HEALTHSOUTH REHABILITATION HOSPITAL   

                    Street Address        City State Code Zip Code  \
0            2505 U.S. HIGHWAY 431        BOAZ         AL   35957-   
1              2525 NORTH BROADWAY   RED LODGE         MT    59806   
2                1231 SOUTH STREET   DADEVILLE         AL    36853   
3              3800 RIDGEWAY DRIVE  BIRMINGHAM         AL    35209   
4  901 NORTH CLEARWATER-LARGO ROAD       LARGO         FL    33770   

       County  Medicare CBSA Number Rural Versus Urban CCN Facility Type  \
0    MARSHALL               13820.0                  R               STH   
1      CARBON               99927.0                  R               CAH   
2  TALLAPOOSA               99919.0                  U               STH   
3   JEFFERSON               13820.0                  U                RH   
4    PINELLAS               45300.0                  U                RH   

   Provider Type  Type of Control Fiscal Year Begin Date Fiscal Year End Date  \
0              1                9             10/01/2010           09/30/2011   
1              1                2             01/01/2011           12/31/2011   
2              1                5             01/01/2011           12/31/2011   
3              5                4             01/01/2011           12/31/2011   
4              5                4             01/01/2011           12/31/2011   

   FTE - Employees on Payroll  Number of Interns and Residents (FTE)  \
0                      598.72                                    NaN   
1                       70.56                                    NaN   
2                       72.65                                    NaN   
3                      297.77                                    NaN   
4                      154.60                                    NaN   

   Total Days Title V  Total Days Title XVIII  Total Days Title XIX  \
0                 NaN                  9132.0                4591.0   
1                 NaN                   922.0                  10.0   
2                 NaN                  1564.0                 299.0   
3                 NaN                 23060.0                 230.0   
4                 NaN                 13304.0                 699.0   

   Total Days (V + XVIII + XIX + Unknown)  Number of Beds  \
0                                 19641.0           114.0   
1                                  1196.0            25.0   
2                                  2681.0            46.0   
3                                 32378.0           100.0   
4                                 16961.0            70.0   

   Total Bed Days Available  Total Discharges Title V  \
0                   41610.0                       NaN   
1                    9125.0                       NaN   
2                   16790.0                       NaN   
3                   36500.0                       NaN   
4                   25550.0                       NaN   

   Total Discharges Title XVIII  Total Discharges Title XIX  \
0                        2109.0                      1134.0   
1                         104.0                         3.0   
2                         468.0                        75.0   
3                        1668.0                        25.0   
4                        1086.0                        44.0   

   Total Discharges (V + XVIII + XIX + Unknown)  \
0                                        5283.0   
1                                         155.0   
2                                         892.0   
3                                        2390.0   
4                  

In [22]:
# Display the last 5 rows of the dataset
hospital_provider_cost_data.tail()

rpt_rec_num  Provider CCN                         Hospital Name  \
73969       776749        251325             COVINGTON COUNTY HOSPITAL   
73970       776764        330273                PUTNAM HOSPITAL CENTER   
73971       776821        453304              TEXAS CHILDRENS HOSPITAL   
73972       776830        520096   WHEATON FRANCISCAN HEALTHCARE - ALL   
73973       776832        520136  ASCENSION SE WISCONSIN HOSPITAL  INC   

                  Street Address       City State Code Zip Code     County  \
73969   803 GERALD MCRANEY DRIVE    COLLINS         MS    39428  COVINGTON   
73970         670 STONELEIGH AVE     CARMEL         NY    10512     PUTNAM   
73971                6621 FANNIN    HOUSTON         TX    77030     HARRIS   
73972         3801 SPRING STREET     RACINE         WI   53405-     RACINE   
73973  5000 WEST CHAMBERS STREET  MILWAUKEE         WI   53210-  MILWAUKEE   

       Medicare CBSA Number Rural Versus Urban CCN Facility Type  \
73969               25620.0                  U               CAH   
73970               35614.0                  U               STH   
73971               26420.0                  U                CH   
73972               39540.0                  U               STH   
73973               33340.0                  U               STH   

       Provider Type  Type of Control Fiscal Year Begin Date  \
73969              1                9             10/01/2021   
73970              1                2             10/01/2021   
73971              7                2             10/01/2021   
73972              1                1             07/01/2022   
73973              1                1             07/01/2022   

      Fiscal Year End Date  FTE - Employees on Payroll  \
73969           09/30/2022                      263.92   
73970           09/30/2022                      485.43   
73971           09/30/2022                    11381.06   
73972           06/30/2023                     1007.11   
73973           06/30/2023                     1119.01   

       Number of Interns and Residents (FTE)  Total Days Title V  \
73969                                    NaN                 NaN   
73970                                   6.79                 NaN   
73971                                 471.86                 NaN   
73972                                   8.49                 NaN   
73973                                  42.57                 NaN   

       Total Days Title XVIII  Total Days Title XIX  \
73969                  3316.0                   1.0   
73970                  6281.0                 460.0   
73971                   573.0               20313.0   
73972                  7488.0                3179.0   
73973                 10424.0                8085.0   

       Total Days (V + XVIII + XIX + Unknown)  Number of Beds  \
73969                                  4872.0            25.0   
73970                                 12328.0            81.0   
73971                                248901.0           863.0   
73972                                 41292.0           285.0   
73973                                 51396.0           301.0   

       Total Bed Days Available  Total Discharges Title V  \
73969                    9125.0                       NaN   
73970                   29487.0                       NaN   
73971                  315279.0                       NaN   
73972                  104025.0                       NaN   
73973                  109865.0                       NaN   

       Total Discharges Title XVIII  Total Discharges Title XIX  \
73969                          64.0                        11.0   
73970                        1390.0                        82.0   
73971                          83.0                      1635.0   
73972                        1542.0                       577.0   
73973                        2060.0                       703.0   

       Total Discharges (V + XVIII + XIX + Unknow

You can see that the dataset contains a mix of categorical, geographical, and numerical variables.

Each record (row) corresponds to a hospital’s annual Medicare cost report submission for a specific fiscal year. The dataset captures:

- Operational statistics (beds, employees, interns/residents)
- Utilization metrics (patient days, discharges, inpatient vs. outpatient activity)
- Financial performance (costs, charges, revenues, charity care, uncompensated care, net income)
- Geographical details (city, state, county, rural vs. urban classification)
- Organizational attributes (facility type, ownership/control type, teaching status)

Next, we will use df.shape and df.info() to explore the dataset’s dimensions and variable types in more detail.

### Data Validation

In [32]:
# Shape of the data set (rows, columns)
print("Shape of dataset:", hospital_provider_cost_data.shape)

Shape of dataset: (73974, 118)


In [39]:
# Summarised information of data set
hospital_provider_cost_data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 73974 entries, 0 to 73973
Columns: 118 entries, rpt_rec_num to Cost Report Year
dtypes: float64(103), int64(5), object(10)
memory usage: 66.6+ MB


There are 73,974 rows including header and 118 columns. Data type is correct and matches the corresponding values.

In [44]:
# Column names and data types
print("\nColumn Info:")
print(hospital_provider_cost_data.dtypes)


Column Info:
rpt_rec_num                            int64
Provider CCN                           int64
Hospital Name                         object
Street Address                        object
City                                  object
                                      ...   
Net Revenue from Medicaid            float64
Medicaid Charges                     float64
Net Revenue from Stand-Alone CHIP    float64
Stand-Alone CHIP Charges             float64
Cost Report Year                       int64
Length: 118, dtype: object


In [50]:
# Quick descriptive stats (numeric only)
print("\nDescriptive Statistics (numeric columns):")
display(hospital_provider_cost_data.describe().T)


Descriptive Statistics (numeric columns):


count      mean       std       min  \
rpt_rec_num                        73974.0  6.78e+05  8.53e+04  2.85e+02   
Provider CCN                       73974.0  2.69e+05  1.58e+05  1.00e+04   
Medicare CBSA Number               73048.0  5.37e+04  3.42e+04  1.00e+00   
Provider Type                      73974.0  1.72e+00  1.55e+00  1.00e+00   
Type of Control                    73974.0  4.23e+00  3.22e+00  1.00e+00   
...                                    ...       ...       ...       ...   
Net Revenue from Medicaid          54482.0  2.38e+07  1.61e+08 -2.69e+07   
Medicaid Charges                   54457.0  1.27e+08  2.92e+08 -2.12e+07   
Net Revenue from Stand-Alone CHIP  12037.0  7.81e+05  4.38e+06 -9.51e+04   
Stand-Alone CHIP Charges           12218.0  3.75e+06  1.77e+07 -1.05e+06   
Cost Report Year                   73974.0  2.02e+03  3.44e+00  2.01e+03   

                                        25%       50%       75%       max  
rpt_rec_num                        6.19e+05  6.96e+05  7.48e+05  7.77e+05  
Provider CCN                       1.40e+05  2.60e+05  3.92e+05  7.13e+05  
Medicare CBSA Number               2.64e+04  3.97e+04  9.99e+04  1.00e+05  
Provider Type                      1.00e+00  1.00e+00  1.00e+00  1.10e+01  
Type of Control                    2.00e+00  4.00e+00  5.00e+00  1.30e+01  
...                                     ...       ...       ...       ...  
Net Revenue from Medicaid          1.86e+06  6.45e+06  2.01e+07  3.11e+10  
Medicaid Charges                   6.79e+06  3.26e+07  1.24e+08  1.57e+10  
Net Revenue from Stand-Alone CHIP  2.53e+04  9.40e+04  3.58e+05  3.00e+08  
Stand-Alone CHIP Charges           1.17e+05  4.96e+05  1.96e+06  8.85e+08  
Cost Report Year                   2.01e+03  2.02e+03  2.02e+03  2.02e+03  

[108 rows x 8 columns]

In [55]:
# Quick descriptive stats (categorical only)
print("\nDescriptive Statistics (categorical columns):")
display(hospital_provider_cost_data.describe(include=['O']).T)


Descriptive Statistics (categorical columns):


count unique                                   top  \
Hospital Name           73974   9118  ENCOMPASS HEALTH REHABILITATION HOSP   
Street Address          73916   8543                    444 LAFAYETTE ROAD   
City                    73974   3418                               HOUSTON   
State Code              73974     56                                    TX   
Zip Code                73974   6795                            55164-0979   
County                  70162   1899                           LOS ANGELES   
Rural Versus Urban      73048      2                                     U   
CCN Facility Type       73974      8                                   STH   
Fiscal Year Begin Date  73974    955                            01/01/2022   
Fiscal Year End Date    73974    746                            12/31/2021   

                         freq  
Hospital Name             485  
Street Address            102  
City                      506  
State Code               7123  
Zip Code                  102  
County                   1180  
Rural Versus Urban      44409  
CCN Facility Type       40901  
Fiscal Year Begin Date   2456  
Fiscal Year End Date     2457

In [59]:
# Count missing values per column
print("\nMissing Values per Column:")
print(hospital_provider_cost_data.isnull().sum())


Missing Values per Column:
rpt_rec_num                              0
Provider CCN                             0
Hospital Name                            0
Street Address                          58
City                                     0
                                     ...  
Net Revenue from Medicaid            19492
Medicaid Charges                     19517
Net Revenue from Stand-Alone CHIP    61937
Stand-Alone CHIP Charges             61756
Cost Report Year                         0
Length: 118, dtype: int64


In [62]:
# Percentage of missing values
print("\nPercentage Missing:")
print((hospital_provider_cost_data.isnull().sum() / len(hospital_provider_cost_data) * 100).round(2))


Percentage Missing:
rpt_rec_num                           0.00
Provider CCN                          0.00
Hospital Name                         0.00
Street Address                        0.08
City                                  0.00
                                     ...  
Net Revenue from Medicaid            26.35
Medicaid Charges                     26.38
Net Revenue from Stand-Alone CHIP    83.73
Stand-Alone CHIP Charges             83.48
Cost Report Year                      0.00
Length: 118, dtype: float64


In [64]:
# Detect potential outliers using IQR
numeric_cols = hospital_provider_cost_data.select_dtypes(include=[np.number]).columns
Q1 = hospital_provider_cost_data[numeric_cols].quantile(0.25)
Q3 = hospital_provider_cost_data[numeric_cols].quantile(0.75)
IQR = Q3 - Q1
outliers = ((hospital_provider_cost_data[numeric_cols] < (Q1 - 1.5 * IQR)) | (hospital_provider_cost_data[numeric_cols] > (Q3 + 1.5 * IQR))).sum()

print("\nPotential Outliers per Numeric Column:")
print(outliers)


Potential Outliers per Numeric Column:
rpt_rec_num                            247
Provider CCN                             0
Medicare CBSA Number                     0
Provider Type                        17411
Type of Control                       8866
                                     ...  
Net Revenue from Medicaid             6224
Medicaid Charges                      6070
Net Revenue from Stand-Alone CHIP     1700
Stand-Alone CHIP Charges              1579
Cost Report Year                         0
Length: 108, dtype: int64


## Exploratory Data Analysis (EDA)

## Key Findings / Insights

## Limitations

## Next Steps / Future Work

## Notebook Outline
1. Data Import & Inspection
2. Data Cleaning
3. Exploratory Data Analysis (EDA)
4. Visualizations
5. Insights & Conclusions